pkg/config/config.go

```diff
@@ -168,31 +199,48 @@ func (gc *Config) addSecretDataToConfig(secret *v1.Secret) error {
 	return gc.addDataToConfig(extractedValues)
 }
 
-func (gc *Config) addDataToConfig(data map[string]string) error {
-	gc.lock.Lock()
-	defer gc.lock.Unlock()
+func (gc *Config) addDataToConfig(rawData map[string]string) error {
+	data := configData{
+		caCerts: rawData["caCerts"],
+		proxyOpts: ProxyOpts{
+			HTTPProxy:  rawData["httpProxy"],
+			HTTPSProxy: rawData["httpsProxy"],
+			NoProxy:    gc.replaceServiceHostPlaceholder(rawData["noProxy"]),
+		},
+		skipTLSVerify: rawData["dangerousSkipTLSVerify"],
+	}
+
+	if val := rawData["appDefaultSyncPeriod"]; len(val) > 0 {
+		dur, err := time.ParseDuration(val)
+		if err != nil {
+			return fmt.Errorf("Unmarshaling appDefaultSyncPeriod as duration: %s", err)
+		}
+		data.appDefaultSyncPeriod = dur
+	}
 
-	gc.caCerts = data["caCerts"]
-	gc.proxyOpts = ProxyOpts{
-		HTTPProxy:  data["httpProxy"],
-		HTTPSProxy: data["httpsProxy"],
-		NoProxy:    gc.replaceServiceHostPlaceholder(data["noProxy"]),
+	if val := rawData["appMinimumSyncPeriod"]; len(val) > 0 {
+		dur, err := time.ParseDuration(val)
+		if err != nil {
+			return fmt.Errorf("Unmarshaling appMinimumSyncPeriod as duration: %s", err)
+		}
+		data.appMinimumSyncPeriod = dur
 	}
 
-	if val := data["kappDeployRawOptions"]; len(val) > 0 {
+	if val := rawData["kappDeployRawOptions"]; len(val) > 0 {
 		var opts []string
 		err := json.Unmarshal([]byte(val), &opts)
 		if err != nil {
 			return fmt.Errorf("Unmarshaling kappDeployRawOptions as JSON: %s", err)
 		}
 		// Allowed flags will be verified before kapp is invoked within Kapp class.
 		// (See pkg/deploy/kapp_restrict.go).
-		gc.kappDeployRawOptions = opts
-	} else {
-		gc.kappDeployRawOptions = nil
+		data.kappDeployRawOptions = opts
 	}
 
-	gc.skipTLSVerify = data["dangerousSkipTLSVerify"]
+	gc.dataLock.Lock()
```

Comment:
would it still be safer to have the `defer gc.datalock.unlock` call immediately after you get the lock?   I realize currently there's only a single assignment in between and it seems like either that would cause some deep panic, or not,  but is it more 'future-friendly' somehow?

Reply:
sure can change back (dont have strong opinion). commit that introduced the change explicitly forces us to change whole struct as a value, rather than individual things so in theory this line should not change for a long time.

Reply:
done